core/modules/comment/comment.module

```diff
@@ -1165,6 +1175,19 @@ function comment_form_node_type_form_alter(&$form, $form_state) {
       '#default_value' => $node_type->settings['comment_close_days'],
       '#field_suffix' => t('days'),
     );
+    $form['comment']['comment_title_options'] = array(
+      '#type' => 'radios',
+      '#title' => t('Comment titles'),
+      '#default_value' => $node_type->settings['comment_title_options'],
+      '#options' => array(
+        COMMENT_TITLE_CUSTOM => t('Custom'),
+        COMMENT_TITLE_AUTO => t('Auto generated'),
+        COMMENT_TITLE_HIDDEN => t('Hidden'),
+      ),
+    );
+    $form['comment']['comment_title_options'][COMMENT_TITLE_CUSTOM]['#description'] = t('Comment titles are visible. A title field is added to comment forms to allow for custom titles.');
+    $form['comment']['comment_title_options'][COMMENT_TITLE_AUTO]['#description'] = t('Comment titles are visible and are automatically generated from the comment body.');
+    $form['comment']['comment_title_options'][COMMENT_TITLE_HIDDEN]['#description'] = t('Comment titles are hidden (this option might not be supported by some themes).');
```

Comment:
I don't think that the "Comment titles are hidden" text provides any value/clarification for the "Hidden" option. I also don't think that repeating "Comment titles are visible." in the other two options is ideal - it could instead be incorporated in the radio option, which would also make the distinction be tween the 2 "visible" options and the "hidden" one. Something like this perhaps?:

```suggestion
    $form['comment']['comment_title_options'] = array(
      '#type' => 'radios',
      '#title' => t('Comment titles'),
      '#default_value' => $node_type->settings['comment_title_options'],
      '#options' => array(
        COMMENT_TITLE_CUSTOM => t('Visible, custom'),
        COMMENT_TITLE_AUTO => t('Visible, auto-generated'),
        COMMENT_TITLE_HIDDEN => t('Hidden'),
      ),
    );
    $form['comment']['comment_title_options'][COMMENT_TITLE_CUSTOM]['#description'] = t('A title field is provided, to allow customizing comment titles.');
    $form['comment']['comment_title_options'][COMMENT_TITLE_AUTO]['#description'] = t('Comment titles are automatically generated from the comment body.');
    $form['comment']['comment_title_options'][COMMENT_TITLE_HIDDEN]['#description'] = t('Note: this option might not be supported by some themes.');
```

Reply:
...for visual comparison:

Before:
![image](https://user-images.githubusercontent.com/2423362/185806999-e3f2bc07-463d-48d3-86f5-39fb2fe377c6.png)

After:
![image](https://user-images.githubusercontent.com/2423362/185807063-aae58e55-8721-4a29-b07b-f4a9429bd3f4.png)

Reply:
IMHO - Good ideas. Added.